cdc/api/v2/api_test.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/stretchr/testify/require"
+	"github.com/tikv/client-go/v2/oracle"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+type mockStatusProvider struct {
+	mock.Mock
+}
+
+// MockPDClient mocks pd.Client to facilitate unit testing.
+type MockPDClient struct {
+	pd.Client
+}
+
+func (m *MockPDClient) GetTS(ctx context.Context) (int64, int64, error) {
+	return oracle.GetPhysical(time.Now()), 0, nil
+}
+
+func (p *mockStatusProvider) GetAllChangeFeedStatuses(ctx context.Context) (map[model.ChangeFeedID]*model.ChangeFeedStatus, error) {
+	args := p.Called(ctx)
+	return args.Get(0).(map[model.ChangeFeedID]*model.ChangeFeedStatus), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetChangeFeedStatus(ctx context.Context, changefeedID model.ChangeFeedID) (*model.ChangeFeedStatus, error) {
+	args := p.Called(ctx, changefeedID)
+	log.Info("err", zap.Error(args.Error(1)))
+	return args.Get(0).(*model.ChangeFeedStatus), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetAllChangeFeedInfo(ctx context.Context) (map[model.ChangeFeedID]*model.ChangeFeedInfo, error) {
+	args := p.Called(ctx)
+	return args.Get(0).(map[model.ChangeFeedID]*model.ChangeFeedInfo), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetChangeFeedInfo(ctx context.Context, changefeedID model.ChangeFeedID) (*model.ChangeFeedInfo, error) {
+	args := p.Called(ctx)
+	return args.Get(0).(*model.ChangeFeedInfo), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetAllTaskStatuses(ctx context.Context, changefeedID model.ChangeFeedID) (map[model.CaptureID]*model.TaskStatus, error) {
+	args := p.Called(ctx)
+	return args.Get(0).(map[model.CaptureID]*model.TaskStatus), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetTaskPositions(ctx context.Context, changefeedID model.ChangeFeedID) (map[model.CaptureID]*model.TaskPosition, error) {
+	args := p.Called(ctx)
+	return args.Get(0).(map[model.CaptureID]*model.TaskPosition), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetProcessors(ctx context.Context) ([]*model.ProcInfoSnap, error) {
+	args := p.Called(ctx)
+	return args.Get(0).([]*model.ProcInfoSnap), args.Error(1)
+}
+
+func (p *mockStatusProvider) GetCaptures(ctx context.Context) ([]*model.CaptureInfo, error) {
+	args := p.Called(ctx)
+	return args.Get(0).([]*model.CaptureInfo), args.Error(1)
+}
+
+type testCase struct {
+	url    string
+	method string
+}
+
+func newRouter(c *capture.Capture, p *mockStatusProvider) *gin.Engine {
+	router := gin.New()
+	RegisterOpenAPIV2Routes(router, NewOpenAPIV2ForTest(c, p))
+	return router
+}
+
+func newStatusProvider() *mockStatusProvider {
+	statusProvider := &mockStatusProvider{}
+	return statusProvider
+}
+
+func TestGetTso(t *testing.T) {
+	t.Parallel()
+	mockPDClient := &MockPDClient{}
+	mockManager := upstream.NewManager4Test(mockPDClient)
+	cp := capture.NewCaptureWithManager4Test(mockManager)
+	router := newRouter(cp, newStatusProvider())
+	w := httptest.NewRecorder()
+
+	api := testCase{url: "/api/v2/tso", method: "GET"}
```

Comment:
Why is a variable of type `testCase` is named `api`?

Reply:
ok